### ISSUE TITLE
Naughtyblog & JavGuru not working

### ISSUE THREAD

**Issue by skittLez82** [dobbelina/repository.dobbelina#1475]
### Prerequisites

- [X] I have searched the issue tracker to avoid opening a duplicate issue.
- [X] I have read the [documentation](https://github.com/dobbelina/repository.dobbelina/wiki/Cumination:-Debug-Logging-and-Uploading) on how to create a debug log file.

### Description

Naughtyblog gives error, "It looks like this website is down." and JavGuru gives error, "This site has a Cloudflare Challenge,"

### Kodi version

21.1

### Cumination version

1.1.123

### Operating System and version

Shield Android TV Version: 9.1.1(33.2.0.157)

### URL to debug log file

https://paste.kodi.tv/xefebipuni

**Comment by donutmiddles** [comment]
Duplicate of https://github.com/dobbelina/repository.dobbelina/issues/1455.

See that most recent screenshot someone else posted where it shows the RD site and "infringing file" error. It's because of the changes RD made here, https://torrentfreak.com/real-debrid-implements-extreme-anti-piracy-filters-to-appease-film-companies-241122/

Some other addons have been reworked to "fix" this and just not point to the immediateAvailability API call, but it will require some rewriting here.

**Comment by Skates1616** [comment]
SeaPorn has a similar issue, all links fail with RD.

**Comment by EvelK1203** [comment]
SeaPorn and Naughtyblog use Rapidgator for download links. Real Debrid are not allowing any RG links for video to be cached on their servers. Other addons work arounds are for torrents, I don't think there is anything that can be done on the Kodi end. Can only hope that RD change their minds. Of note Premiumize are also having issues with RG links.

JavGuru issue is known and waiting on an update from Flaresolverr, https://github.com/dobbelina/repository.dobbelina/issues/1436

**Comment by Mulanzone** [comment]
I tried this add-on in Kodi on Shield and got this error with RD account.  Tried on my Mac, and also got the error.  When I tried on PC it worked.  RG links dont work but the other ones do.   Is there a way to set the User-Agent manually to see if that would fix the issue?  Maybe allow the option to choose the User-Agent in settings if it does help?  I had read elsewhere that the issue might be due to that.  Thanks

**Comment by 12asdfg12** [comment]
The user-agent is stored in the `settings.xml` file in the `userdata\addon_data\plugin.video.cumination` folder.
You can edit this file and modify the following line:
`<setting id="current_ua" default="true">_your_user_agent_</setting>`

However, I doubt that changing the user-agent will resolve the issue.

**Comment by Gujal00** [comment]
RD is accessed through ResolveURL and the User-Agent ResolveURL uses is fixed regardless of platform.

**Comment by Cgritsch89** [comment]
When will fix this bug?

**Comment by Gujal00** [comment]
It is not a bug, RD flags anything from rapidgator as infringing so cant use. Use another debrid service

**Comment by Mulanzone** [comment]
Interesting.  When opening Naughtyblog links on a Windows Kodi install, streams that have a K2S link sometimes work, though RG links don’t work.   When viewing the same streams on Shield or Mac, Kodi automatically says the Website is down even though it works on Windows.

**Comment by MohamadFtah** [comment]
> It is not a bug, RD flags anything from rapidgator as infringing so cant use. Use another debrid service

do you know what debrid service works?
